Content.Server/Doors/Systems/DoorSystem.cs

```diff
@@ -91,124 +92,128 @@ protected override void PlaySound(EntityUid uid, string sound, AudioParams audio
+private void OnWeldCancelled(EntityUid uid, DoorComponent door, WeldCancelledEvent args)
+{
+    door.BeingWelded = false;
+}
 
-    private void OnWeldFinished(EntityUid uid, DoorComponent door, WeldFinishedEvent args)
-    {
-        door.BeingWelded = false;
+private void OnWeldFinished(EntityUid uid, DoorComponent door, WeldFinishedEvent args)
+{
+    door.BeingWelded = false;
 
-        if (!door.Weldable)
-            return;
+    if (!door.Weldable)
+        return;
 
-        if (door.State == DoorState.Closed)
-            SetState(uid, DoorState.Welded, door);
-        else if (door.State == DoorState.Welded)
-            SetState(uid, DoorState.Closed, door);
-    }
+    if (door.State == DoorState.Closed)
+        SetState(uid, DoorState.Welded, door);
+    else if (door.State == DoorState.Welded)
+        SetState(uid, DoorState.Closed, door);
+}
 
-    private void OnPryCancelled(EntityUid uid, DoorComponent door, PryCancelledEvent args)
-    {
-        door.BeingPried = false;
-    }
+private void OnPryCancelled(EntityUid uid, DoorComponent door, PryCancelledEvent args)
+{
+    door.BeingPried = false;
+}
 
-    private void OnPryFinished(EntityUid uid, DoorComponent door, PryFinishedEvent args)
-    {
-        door.BeingPried = false;
+private void OnPryFinished(EntityUid uid, DoorComponent door, PryFinishedEvent args)
+{
+    door.BeingPried = false;
 
-        if (door.State == DoorState.Closed)
-            StartOpening(uid, door);
-        else if (door.State == DoorState.Open)
-            StartClosing(uid, door);
-    }
-    #endregion
+    if (door.State == DoorState.Closed)
+        StartOpening(uid, door);
+    else if (door.State == DoorState.Open)
+        StartClosing(uid, door);
+}
+#endregion
 
-    /// <summary>
-    ///     Does the user have the permissions required to open this door?
-    /// </summary>
-    public override bool HasAccess(EntityUid uid, EntityUid? user = null, AccessReaderComponent? access = null)
-    {
-        // TODO network AccessComponent for predicting doors
+/// <summary>
+///     Does the user have the permissions required to open this door?
+/// </summary>
+public override bool HasAccess(EntityUid uid, EntityUid? user = null, AccessReaderComponent? access = null)
+{
+    // TODO network AccessComponent for predicting doors
 
-        // if there is no "user" we skip the access checks. Access is also ignored in some game-modes.
-        if (user == null || AccessType == AccessTypes.AllowAll)
-            return true;
+    // if there is no "user" we skip the access checks. Access is also ignored in some game-modes.
+    if (user == null || AccessType == AccessTypes.AllowAll)
+        return true;
+
+    if (!Resolve(uid, ref access, false))
+        return true;
 
-        if (!Resolve(uid, ref access, false))
+    // If the door is on emergency access we skip the checks.
+    if (TryComp<AirlockComponent>(uid, out var airlock) && airlock.EmergencyAccess)
             return true;
```

Comment:
This should just be before the `access` resolve, seeing as `access` isn't needed if this is the case.

Reply:
Done